dashboard-new/src/app/(authenticated)/components/searchAction.ts

```diff
@@ -48,15 +49,15 @@ const bookmarkFrom = (bookmark?: string): Buffer | undefined => {
   return Buffer.from(bookmark, 'base64')
 }
 
-interface SearchResult {
+export interface SearchResult {
   type: SearchType
   bookmark?: string
   results: any
```

Comment:
Are there generics in Javascript? If so, this would be a good use-case for it. Alternatively, you could just pass the proto classes around.

Reply:
Yes, that's actually what I'm doing. Although not for this interface because I'm using that just as the parent interface, then below this I'm mapping which `SearchType` belongs to each `result` type. That is because I'm using a generic search function which resolves the type.